umbral/utils.py

```diff
@@ -42,3 +44,23 @@ def kdf(ecpoint, key_length):
 
 def get_curve_keysize_bytes(curve):
     return int(math.ceil(curve.key_size / 8.00))
+
+
+class AbstractFromToBytes(ABC):
+    @staticmethod
+    @abstractmethod
+    def from_bytes(cls, data: bytes, curve: ec.EllipticCurve = None):
```

Comment:
I have some problems with this part... The `curve` parameter only makes sense in some classes (e.g, BigNum, Point), but not in others (e.g., KFrag). Being an optional parameter makes this problem not very relevant, but at the same time it means we have to explicitly throw an error or something similar when it is used incorrectly (e.g., inputing a `curve` when deserializing a KFrag).

Reply:
well, what if we abstract away all the arguments? Then each derived class should know what to do, since this is not an implementation, just a way to force an implementation, e.g.

```python
class AbstractFromTo(ABC):
  @staticmethod
  @abstractmethod
  def from_bytes(*args, **kwargs):
    pass
```
and the same with the other methods

Reply:
BTW, I just saw that the `Fragments` breaks the `from_bytes` format used in the other classes, [code here](https://github.com/nucypher/pyUmbral/pull/105/files/bc586efe57fc80a9c472224d00d59fd109562afe#diff-bce02964095b01c806e4cc506eda1ee0L21):

```python
class KFrag(AbstractFromToBytes):
  ...
    @staticmethod
    def from_bytes(data: bytes, curve: ec.EllipticCurve = None):
        pass
```
while all other versions use a `@classmethod`. 
Do you want to keep consistency `from_bytes` and `to_bytes` or it was written as such for a specific reason?

Reply:
Agreed - I cannot think of any reason why `from_bytes` is not a `@classmethod`.